app/uk/gov/hmrc/personaldetailsvalidation/PersonalDetailsValidationResourceController.scala

```diff
@@ -47,10 +47,19 @@ class PersonalDetailsValidationResourceController @Inject()(personalDetailsValid
   def create = Action.async(parse.json) { implicit request =>
     withJsonBody[PersonalDetails] { personalDetails =>
 
-      def handleMatchingDone(validationId: ValidationId) =
+      def handleMatchingDone(validationId: ValidationId) = {
         Future.successful(Created.withHeaders(LOCATION -> routes.PersonalDetailsValidationResourceController.get(validationId).url))
+      }
 
-      def handleException(exception: Exception): Future[Result] = Future.failed(exception)
+      def handleException(exception: Exception): Future[Result] = {
```

Comment:
This code can be avoided if validations are done as part of Reads.

Reply:
👍